### PR TITLE
Add HTTPS emitter plugin

### PR DESCRIPTION
**Pull request by tatsuhirochiba**
This PR provides `https` emitter plugin. 
It resolves EmitterUnsupportedProtocol error when pushing crawled data to `https` endpoint.

### CODE REVIEW

**Comment by sahilsuneja1** [comment]
Thanks @tatsuhirochiba! 

This seems identical to PR#264.

In either case, since it is identical to the http plugin, we should perhaps deduplicate the https plugin by extending from http plugin class?

Otherwise, if you want to give more functionality to the separate https plugin, should we allow `'cert='` or `verify=<path to certs>'` in `requests.post()`? (Ref: http://docs.python-requests.org/en/latest/user/advanced/#ssl-cert-verification)

Plus, this still needs to incorporate fix #259  

+ @nadgowdas

**Comment by tatsuhirochiba** [comment]
@sahilsuneja1 Oh, I did not notice that the similar PR had already existed. :)
I agree with inheriting common methods from http emitter plugin. I just want to pass `https` protocol. 

I created a base emitter class (`base_http_emitter.py`) from existing `http_emitter`, and changed both of `http/https emitter` to use the base emitter.  (or should we extend http_emitter directly?)

This does not incorporate fix #259.

**Comment by sahilsuneja1** [comment]
Thanks @tatsuhirochiba for the edits. I'll have a closer look later. In the meanwhile, one more request- could you please squash these commits :)

File: crawler/plugins/emitters/base_http_emitter.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                self.post(line, metadata)
+        else:
+            self.post(iostream.getvalue(), metadata)
+
+    def post(self, content='', metadata={}):
+        headers = {'content-type': 'application/csv'}
+        for attempt in range(self.max_retries):
+            try:
+                response = requests.post(self.url, headers=headers,
+                                         params=metadata,
```
Comment (sahilsuneja1) [inline crawler/plugins/emitters/base_http_emitter.py]: if you just remove the `params` parameter you'll address pr #259's issue #251 as well :)

**Comment by tatsuhirochiba** [comment]
@sahilsuneja1 Thanks for review! I included the fix in the code provided from #259.

**Comment by sahilsuneja1** [comment]
@tatsuhirochiba  Thanks a lot! Good to go once you squash the commits into a single one :)

**Comment by nadgowdas** [comment]
@tatsuhirochiba @sahilsuneja1 we have revised PR contribution policies.. Please review it before creating any new PRs..

**Comment by tatsuhirochiba** [comment]
@nadgowdas Sorry for PR conflict.. I have reviewed the new policy, and will review issues before creating a new PR.

**Comment by nadgowdas** [comment]
@tatsuhirochiba no worries:) and thanks for your hard work.

**Pull request merged**